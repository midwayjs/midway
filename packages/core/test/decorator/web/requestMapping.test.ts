import {
  Get,
  Post,
  Options,
  Head,
  All,
  getClassMetadata,
  WEB_ROUTER_KEY,
  RequestMapping,
} from '../../../src';

class Test {
  @Get('/get', { routerName: 'get', middleware: ['hello'] })
  async doGet() {
    // ignore
  }

  @Post('/get', { routerName: 'post', middleware: ['hello'], summary: 'test post method' })
  async doPost() {
    // ignore
  }

  @Options('/get', { routerName: 'options', middleware: ['hello'], description: 'test option method' })
  async doOptions() {
    // ignore
  }

  @Head('/get', { routerName: 'head', middleware: ['hello'] })
  async doHead() {
    // ignore
  }

  @All(null, { routerName: 'all' })
  async doAll() {
    // ignore
  }

  async ttt() {}
}

describe('/test/web/requestMapping.test.ts', () => {
  it('requestMapping decorator should be ok', () => {
    const meta = getClassMetadata(WEB_ROUTER_KEY, Test);
    expect(meta).toMatchSnapshot();

    const dd = RequestMapping();
    dd(Test, 'ttt', null);

    const metadd = getClassMetadata(WEB_ROUTER_KEY, Test);
    expect(metadd[metadd.length - 1]).toMatchSnapshot();

    const bb = RequestMapping({
      requestMethod: null,
    });

    bb(Test, 'ttt', null);
    const metabb = getClassMetadata(WEB_ROUTER_KEY, Test);
    expect(metabb[metabb.length - 1]).toMatchSnapshot();
  });
});
