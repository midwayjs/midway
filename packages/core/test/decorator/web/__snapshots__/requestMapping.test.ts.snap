// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/test/web/requestMapping.test.ts requestMapping decorator should be ok 1`] = `
[
  {
    "description": "",
    "ignoreGlobalPrefix": false,
    "method": "doGet",
    "middleware": [
      "hello",
    ],
    "path": "/get",
    "requestMethod": "get",
    "routerName": "get",
    "summary": "",
  },
  {
    "description": "",
    "ignoreGlobalPrefix": false,
    "method": "doPost",
    "middleware": [
      "hello",
    ],
    "path": "/get",
    "requestMethod": "post",
    "routerName": "post",
    "summary": "test post method",
  },
  {
    "description": "test option method",
    "ignoreGlobalPrefix": false,
    "method": "doOptions",
    "middleware": [
      "hello",
    ],
    "path": "/get",
    "requestMethod": "options",
    "routerName": "options",
    "summary": "",
  },
  {
    "description": "",
    "ignoreGlobalPrefix": false,
    "method": "doHead",
    "middleware": [
      "hello",
    ],
    "path": "/get",
    "requestMethod": "head",
    "routerName": "head",
    "summary": "",
  },
  {
    "description": "",
    "ignoreGlobalPrefix": false,
    "method": "doAll",
    "middleware": undefined,
    "path": "/",
    "requestMethod": "all",
    "routerName": "all",
    "summary": "",
  },
]
`;

exports[`/test/web/requestMapping.test.ts requestMapping decorator should be ok 2`] = `
{
  "description": "",
  "ignoreGlobalPrefix": false,
  "method": "ttt",
  "middleware": [],
  "path": "/",
  "requestMethod": "get",
  "routerName": null,
  "summary": "",
}
`;

exports[`/test/web/requestMapping.test.ts requestMapping decorator should be ok 3`] = `
{
  "description": "",
  "ignoreGlobalPrefix": false,
  "method": "ttt",
  "middleware": undefined,
  "path": "/",
  "requestMethod": "get",
  "routerName": undefined,
  "summary": "",
}
`;
